File: javaSource/BroadcastSample/app/src/main/java/com/guagua/broadcastsample/NetworkChangeReceiver.java
package com.guagua.broadcastsample;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.util.Log;

/**
 * @UpdateRemark: 更新说明：
 * @Version: 1.0
 */
public class NetworkChangeReceiver extends BroadcastReceiver {
    private static final String TAG = "NetworkChangeReceiver";

    @Override
    public void onReceive(Context context, Intent intent) {
        ConnectivityManager manager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo info = manager.getActiveNetworkInfo();
        if (null != info && info.isAvailable()) {
            Log.i(TAG, " 网络可用 O(∩_∩)O ");
        } else {
            Log.i(TAG, " 网络被外星人劫持了 ╮(╯_╰)╭ ");
        }
    }
}

File: kotlinSource/BroadcastSample/app/src/main/java/com/guagua/broadcastsample/NetworkChangeReceiver.kt
package com.guagua.broadcastsample

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.net.ConnectivityManager
import android.net.NetworkInfo
import android.util.Log

/**
 *
 * @ProjectName:    BroadcastSample
 * @Package:        com.guagua.broadcastsample
 * @ClassName:      NetworkChangeReceiver
 * @Description:     java类作用描述
 * @UpdateRemark:   更新说明：
 * @Version:        1.0
 */
class NetworkChangeReceiver : BroadcastReceiver() {
    companion object {
        private const val TAG = "NetworkChangeReceiver"
    }

    override fun onReceive(context: Context?, p1: Intent?) {
        val manager: ConnectivityManager =
            context?.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val info: NetworkInfo? = manager?.activeNetworkInfo
        if (info?.isAvailable == true) {
            Log.i(TAG, " 网络可用 O(∩_∩)O ")
        } else {
            Log.i(TAG, " 网络被外星人劫持了 ╮(╯_╰)╭ ")
        }
    }
}